sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/NullPointerDereference.cs

```diff
@@ -625,4 +625,19 @@ public void Guarded(string s1)
         [AttributeUsage(AttributeTargets.Parameter)]
         public sealed class ValidatedNotNullAttribute : Attribute { }
     }
+
+    public class ExceptionFilters // https://github.com/SonarSource/sonar-csharp/issues/1324
+    {
+        public void Run(object o)
+        {
+            try
+            {
+                Console.WriteLine(o?.ToString());
+            }
+            catch (Exception) when (o != null)
+            {
+                Console.WriteLine(o.ToString());
```

Comment:
Unit tests for more complicated cases:
* catch block with condition that doesn't guard against null -> non-compliant
* multiple catch blocks, some with conditions that guard against nulls and some that don't

Reply:
To check that the rule is looking at every catch block. None of the tests have multiple catch blocks, so they would all pass even if the rule was only looking at the first/last catch or stopping on the first error.